### ISSUE TITLE
ClassCastException on right join

### ISSUE THREAD

**Issue by burnes** [crate/crate#4595]
**CrateDB version**: 1.0.1

**JVM version**: 1.8

**OS version / environment description**: OSX

**Problem description**:

Right Join with Query on a property of an object column leads to the following error:

```
SQLActionException[ClassCastException: org.apache.lucene.util.BytesRef cannot be cast to java.lang.Long]
```

The left join works, the right join leads to the error.

**Steps to reproduce**:

```
CREATE TABLE a (
  id STRING PRIMARY KEY,
  data OBJECT(STRICT) AS (
    x LONG
  )
);

CREATE TABLE b (
  f_key STRING PRIMARY KEY,
  name STRING
);

INSERT INTO a (id, data) VALUES ('A', {"x"=1}), ('B', {"x"=2}), ('C', {"x"=3});
INSERT INTO b (f_key, name) VALUES ('A', 'Anna'), ('B', 'Barbara');

SELECT * FROM a LEFT JOIN b ON a.id = b.f_key WHERE data['x'] > 0;
SELECT * FROM a RIGHT JOIN b ON a.id = b.f_key WHERE data['x'] > 0;
```

**Comment by celaus** [comment]
Thanks for submitting, we'll take a look!

**Comment by matriv** [comment]
The issue has been identified and solved. The fix will be available with 1.0.2 release.
@burnes Thanks again for reporting!
